### PR TITLE
Adding new FileWatcher implementation for Linux based on Inotify

### PR DESCRIPTION
**Pull request by radupopescu**

### CODE REVIEW

File: cvmfs/file_watcher_inotify.cc [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  poll_set[0].events = POLLHUP | POLLIN;
+  poll_set[0].revents = 0;
+  poll_set[1].fd = inotify_fd_;
+  poll_set[1].events = POLLIN;
+  poll_set[1].revents = 0;
+
+  bool stop = false;
+  while (!stop) {
+    int ready = poll(poll_set, 2, -1);
+    if (ready == -1) {
```
Comment (jblomer) [inline cvmfs/file_watcher_inotify.cc]: This might be ok when `errno == EINTR`

File: cvmfs/file_watcher_inotify.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            event = file_watcher::kRenamed;
+          } else if (inotify_event->mask & IN_ATTRIB) {
+            // Attributes
+            event = file_watcher::kAttributes;
+          } else if (inotify_event->mask & IN_IGNORED) {
+            // Attributes
+            event = file_watcher::kIgnored;
+          }
+          bool clear_handler = true;
+          if (event != file_watcher::kInvalid ||
```
Comment (jblomer) [inline cvmfs/file_watcher_inotify.cc]: Shouldn't this be "event not invalid _and_ event not ignored"?

File: cvmfs/file_watcher_inotify.cc [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    if (poll_set[0].revents & POLLIN) {
+      char buffer[1];
+      ReadPipe(read_pipe, &buffer, 1);
+      LogCvmfs(kLogCvmfs, kLogDebug, "FileWatcherInotify - Stopping.\n");
+      stop = true;
+      continue;
+    }
+
+    const size_t event_size = sizeof(struct inotify_event);
+    const size_t buffer_size = event_size + 100;
```
Comment (jblomer) [inline cvmfs/file_watcher_inotify.cc]: The 100 might be better a constant explaining its meaning.

File: cvmfs/file_watcher_inotify.cc [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+  close(inotify_fd_);
+
+  return true;
+}
+
+void FileWatcherInotify::RegisterFilter(const std::string& file_path,
+                                        EventHandler* handler) {
+  bool done = false;
+  BackoffThrottle throttle(1000, 10000, 50000);
```
Comment (jblomer) [inline cvmfs/file_watcher_inotify.cc]: Could move as constants to the header.

File: cvmfs/file_watcher_inotify.cc (hunk 2 above)
Comment (radupopescu) [reply cvmfs/file_watcher_inotify.cc]: Good catch.

**Pull request merged**